packages/babel-parser/test/fixtures/typescript/interface/export-default/output.json

```diff
@@ -0,0 +1,26 @@
+{
+  "type": "File",
+  "start":0,"end":27,"loc":{"start":{"line":1,"column":0},"end":{"line":1,"column":27}},
+  "errors": [
+    "SyntaxError: 'export default interface' must be followed by an identifier. (1:25)"
+  ],
+  "program": {
+    "type": "Program",
+    "start":0,"end":27,"loc":{"start":{"line":1,"column":0},"end":{"line":1,"column":27}},
+    "sourceType": "module",
+    "interpreter": null,
+    "body": [
+      {
+        "type": "ExportDefaultDeclaration",
+        "start":0,"end":27,"loc":{"start":{"line":1,"column":0},"end":{"line":1,"column":27}},
+        "exportKind": "value",
+        "declaration": {
+          "type": "ObjectExpression",
+          "start":25,"end":27,"loc":{"start":{"line":1,"column":25},"end":{"line":1,"column":27}},
+          "properties": []
+        }
+      }
```

Comment:
I don't like this output. It was already the old output we generated before this PR, but since we are modifying the code that generates it we can use this chance to improve it.

If the interface name is missing, we should still parse it as an interface and set `id` to `null`.

We can modify the `tsParseInterfaceDeclaration` function and move the error checking there, instead of in `parseExportDefaultExpression`:
```js
    tsParseInterfaceDeclaration(
      node: N.TsInterfaceDeclaration,
    ): N.TsInterfaceDeclaration {
      if (this.match(tt.name)) {
        node.id = this.parseIdentifier();
        this.checkLVal(
          node.id,
          "typescript interface declaration",
          BIND_TS_INTERFACE,
        );
      } else {
        node.id = null;
        this.raise(
          this.state.start,
          TSErrors.InvalidInterfaceDeclarationInExportDefault,
        );
      }
```

We can then directly call `tsParseInterfaceDeclaration` instead of `tsParseDeclaration` in the code you modified.